plugins/gateway/kong/plugin.go

```diff
@@ -78,50 +105,69 @@ func (conf Config) Response(kong *pdk.PDK) {
 			return
 		}
 	}
-	if telemetriesAPI == nil {
-		var tlsOptions *common.ClientTLSOptions
-		if conf.EnableTLS {
-			tlsOptions = &common.ClientTLSOptions{
-				RootCAFileName: common.CACertFile,
-			}
-		}
-		apiClient, err := common.NewTelemetryAPIClient(conf.Host, tlsOptions)
-		if err != nil {
-			_ = kong.Log.Err(fmt.Sprintf("Failed to create new api client: %v", err))
-			return
-		}
-		telemetriesAPI = apiClient
-	}
 	telemetry, err := createTelemetry(kong)
 	if err != nil {
 		_ = kong.Log.Err(fmt.Sprintf("Failed to create telemetry: %v", err))
 		return
 	}
 
-	params := operations.NewPostTelemetryParams().WithBody(telemetry)
-
-	_, err = telemetriesAPI.Operations.PostTelemetry(params)
+	err = apiclarityClient.PostTelemetry(telemetry)
 	if err != nil {
 		_ = kong.Log.Err(fmt.Sprintf("Failed to post telemetry: %v", err))
 		return
 	}
 	_ = kong.Log.Info(fmt.Sprintf("Telemetry has been sent: %v", telemetry))
 }
 
+func processNewDiscoveredAPI(kong *pdk.PDK) error {
+	host, err := getHost(kong)
+	if err != nil {
+		return fmt.Errorf("failed to get routed service for processNewDiscoveredAPI: %v", err)
+	}
+	// Check for newDiscoveredApi
+	if !common.Contains(discoveredApis, host) {
+		appendNewDiscoveredAPI(host)
+		hosts := []string{host}
+		if err := apiclarityClient.PostNewDiscoveredAPIs(hosts); err != nil {
+			return fmt.Errorf("failed to send newDiscoveredApi request: %v", err)
+		}
+		_ = kong.Log.Info("Sent PostNewDiscoveredAPIs with success")
+	}
+	return nil
+}
+
+func appendNewDiscoveredAPI(host string) {
+	lock.RLock()
+	defer lock.RUnlock()
+
+	discoveredApis = append(discoveredApis, host)
+}
+
 func shouldTrace(kong *pdk.PDK) (bool, error) {
 	routedService, err := kong.Router.GetService()
 	if err != nil {
-		return false, fmt.Errorf("failed to get routed service: %v", err)
+		return false, fmt.Errorf("failed to get routed service for shouldTrace: %v", err)
 	}
 	host, port, _ := parseKongHost(routedService.Host)
-	if traceSamplingClient.ShouldTrace(host, port) {
+	// Note: here 'host' contain already the namespace. i.e. for service 'catalog' on namespace 'sock-shop', host will be 'catalog.sock-shop'
+	if apiclarityClient.ShouldTrace(host, port) {
 		return true, nil
 	}
 	_ = kong.Log.Info("Ignoring host: %v:%v", host, port)
```

Comment:
Debug mode, I know, it was already there.

Reply:
You talk about the _ = kong.Log.Info("Ignoring host: %v:%v", host, port), right? Ok